src/compact.rs

```diff
@@ -249,7 +256,12 @@ impl<T> MaybeMaxEncodedLen for T {}
 /// Trait that tells you if a given type can be encoded/decoded in a compact way.
 pub trait HasCompact: Sized {
 	/// The compact type; this can be
-	type Type: for<'a> EncodeAsRef<'a, Self> + Decode + From<Self> + Into<Self> + MaybeMaxEncodedLen;
+	type Type: for<'a> EncodeAsRef<'a, Self>
+		+ Decode
+		+ DecodeWithMemTracking
```

Comment:
So only mem tracked types can be compact now? I guess that is fine, since `Compact` should be sealed anyway.

Reply:
Yes, this was also my reasoning.